paddle/fluid/eager/utils.cc

```diff
@@ -298,7 +300,9 @@ std::vector<paddle::experimental::Tensor> EagerUtils::SyncToPtenTensors(
   size_t num = tensors.size();
   res.reserve(num);
   for (size_t i = 0; i < num; i++) {
-    const_cast<EagerTensor*>(&(tensors[i]))->SyncToTensor();
+    if (!tensors[i].initialized()) {
```

Comment:
Don't need this after [PR39376](https://github.com/PaddlePaddle/Paddle/pull/39376)